src/main/scala/is/hail/expr/CM.scala

```diff
@@ -162,6 +162,9 @@ object CM {
   def invokePrimitive4[A,B,C,D,R](f: (A, B, C, D) => R)(a: Code[A], b: Code[B], c: Code[C], d: Code[D])
     (implicit act: ClassTag[A], bct: ClassTag[B], cct: ClassTag[C], dct: ClassTag[D], rct: ClassTag[R]) : CM[Code[R]] =
     foreignFun(f).map(_.invoke[R]("apply", Array[Class[_]](act.runtimeClass, bct.runtimeClass, cct.runtimeClass, dct.runtimeClass), Array(a, b, c, d)))
+  def invokePrimitive5[A,B,C,D,R,O](f: (A, B, C, D, R) => O)(a: Code[A], b: Code[B], c: Code[C], d: Code[D], r:Code[R])
+    (implicit act: ClassTag[A], bct: ClassTag[B], cct: ClassTag[C], dct: ClassTag[D], rct: ClassTag[R], oct: ClassTag[O]) : CM[Code[O]] =
+    foreignFun(f).map(_.invoke[O]("apply", Array[Class[_]](act.runtimeClass, bct.runtimeClass, cct.runtimeClass, dct.runtimeClass, rct.runtimeClass ), Array(a, b, c, d, r )))
```

Comment:
I usually use `R` for the return type, so I'd change `f`'s type to `(A,B,C,D,E) => R`, change the new argument to `e: Code[E]`, update ether class tags, change `rct.runtimeClass` to `ect.runtimeClass` and change `r` to `e` in the final `Array`.